### DISCUSSION TITLE
'NoneType' object has no attribute 'lower'

### DISCUSSION THREAD

**Discussion by i-pushkin** [netbox-community/netbox#12267] [Help Wanted!]
When attempting to retrieve data from the cluster, the following error is displayed
```
<class 'AttributeError'>

'NoneType' object has no attribute 'lower'

Python version: 3.10.10
NetBox version: 3.4.7
```
It is not quite clear how to diagnose such an error

**Comment by candlerb** [comment]
Firstly, you'll need to say exactly what you were doing when you got that error. In particular: what path was in the browser URL bar for the previous page, and what did you click within that page?

Secondly, you'll need to capture the full backtrace. You can do this by setting ADMINS and EMAIL in configuration.py (in which case it will be mailed out to you), or by setting DEBUG=True, in which case you'll be able to see it in the web interface.

**Reply by i-pushkin** [reply]
Thank you for the tip. Here is the result:

<html>
<body>
<h1 style="padding: 0px; margin: 0px; font-weight: normal; color: rgb(0, 0, 0); font-family: sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">AttributeError at /plugins/proxbox/full_update/</h1><pre class="exception_value" style="padding: 0px; margin: 10px 0px; font-size: 1.5em; white-space: pre-wrap; word-break: break-word; font-family: sans-serif; color: rgb(87, 87, 87); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">'NoneType' object has no attribute 'lower'</pre>

Request Method: | GET
-- | --
https://netbox.tickerontest.com/plugins/proxbox/full_update/
4.1.7
AttributeError
'NoneType' object has no attribute 'lower'
/opt/netbox/venv/lib/python3.10/site-packages/netbox_proxbox/proxbox_api/updates/virtual_machine.py, line 396, in interfaces_ips
netbox_proxbox.views.ProxmoxFullUpdate
/opt/netbox/venv/bin/python3
3.10.10
['/opt/netbox/netbox',  '/opt/netbox',  '/opt/netbox/venv/bin',  '/usr/lib/python310.zip',  '/usr/lib/python3.10',  '/usr/lib/python3.10/lib-dynload',  '/opt/netbox/venv/lib/python3.10/site-packages']
Mon, 17 Apr 2023 06:14:56 +0000


</body>
</html>

```Environment:


Request Method: GET
Request URL: https://netbox.tickerontest.com/plugins/proxbox/full_update/

Django Version: 4.1.7
Python Version: 3.10.10
Installed Applications:
['django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'django.contrib.humanize',
 'corsheaders',
 'debug_toolbar',
 'graphiql_debug_toolbar',
 'django_filters',
 'django_tables2',
 'django_prometheus',
 'graphene_django',
 'mptt',
 'rest_framework',
 'social_django',
 'taggit',
 'timezone_field',
 'circuits',
 'dcim',
 'ipam',
 'extras',
 'tenancy',
 'users',
 'utilities',
 'virtualization',
 'wireless',
 'django_rq',
 'drf_yasg',
 'netbox_proxbox.ProxboxConfig']
Installed Middleware:
['graphiql_debug_toolbar.middleware.DebugToolbarMiddleware',
 'django_prometheus.middleware.PrometheusBeforeMiddleware',
 'corsheaders.middleware.CorsMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'django.middleware.security.SecurityMiddleware',
 'netbox.middleware.ExceptionHandlingMiddleware',
 'netbox.middleware.RemoteUserMiddleware',
 'netbox.middleware.LoginRequiredMiddleware',
 'netbox.middleware.DynamicConfigMiddleware',
 'netbox.middleware.APIVersionMiddleware',
 'netbox.middleware.ObjectChangeMiddleware',
 'django_prometheus.middleware.PrometheusAfterMiddleware']



Traceback (most recent call last):
  File "/opt/netbox/venv/lib/python3.10/site-packages/django/core/handlers/exception.py", line 56, in inner
    response = get_response(request)
  File "/opt/netbox/venv/lib/python3.10/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/opt/netbox/venv/lib/python3.10/site-packages/django/views/generic/base.py", line 103, in view
    return self.dispatch(request, *args, **kwargs)
  File "/opt/netbox/venv/lib/python3.10/site-packages/django/contrib/auth/mixins.py", line 109, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/opt/netbox/venv/lib/python3.10/site-packages/django/views/generic/base.py", line 142, in dispatch
    return handler(request, *args, **kwargs)
  File "/opt/netbox/venv/lib/python3.10/site-packages/netbox_proxbox/views.py", line 93, in get
    "virtualmachines_table": VMUpdateResult(proxbox_api.update.all(remove_unused = True)["virtualmachines"]),
  File "/opt/netbox/venv/lib/python3.10/site-packages/netbox_proxbox/proxbox_api/update.py", line 448, in all
    vm_updated = virtual_machine(proxmox_json = px_vm_each)
  File "/opt/netbox/venv/lib/python3.10/site-packages/netbox_proxbox/proxbox_api/update.py", line 278, in virtual_machine
    full_update = vm_full_update(netbox_vm, proxmox_json)
  File "/opt/netbox/venv/lib/python3.10/site-packages/netbox_proxbox/proxbox_api/update.py", line 37, in vm_full_update
    ips_updated = updates.virtual_machine.interfaces_ips(netbox_vm, proxmox_vm)
  File "/opt/netbox/venv/lib/python3.10/site-packages/netbox_proxbox/proxbox_api/updates/virtual_machine.py", line 396, in interfaces_ips
    _mac = interface['mac_address'].lower()

Exception Type: AttributeError at /plugins/proxbox/full_update/
Exception Value: 'NoneType' object has no attribute 'lower'
```

My guess is that the problem is that my netbox installation already has data that is in conflict.

**Reply by candlerb** [reply]
The exception is a bug in a plugin (netbox_proxbox), so you need to either disable that plugin, or raise your problem with the plugin author.

**Reply by i-pushkin** [reply]
Sorry, I made a mistake and wrote the question in the wrong repository. )
